src/include/error.h

```diff
@@ -124,67 +124,141 @@
       "encountered an illegal file format or internal value", (uintmax_t)(v))
 
 /*
- * WT_ERR_ASSERT, WT_RET_ASSERT, WT_ASSERT
- *	Assert an expression, aborting in diagnostic mode and otherwise exiting
- * the function with an error. WT_ASSERT is deprecated, and should be used only
- * where required for performance.
+ * Branch prediction hints. If an expression is likely to return true/false we can use this
+ * information to improve performance at runtime. This is not supported for MSVC compilers.
  */
-#ifdef HAVE_DIAGNOSTIC
-#define WT_ASSERT(session, exp)             \
-    do {                                    \
-        if (!(exp)) {                       \
-            __wt_errx(session, "%s", #exp); \
-            __wt_abort(session);            \
-        }                                   \
-    } while (0)
-#define WT_ERR_ASSERT(session, exp, v, ...)    \
-    do {                                       \
-        if (!(exp)) {                          \
-            __wt_err(session, v, __VA_ARGS__); \
-            __wt_abort(session);               \
-        }                                      \
+#if !defined(_MSC_VER)
+#define LIKELY(x) __builtin_expect(!!(x), 1)
+#define UNLIKELY(x) __builtin_expect(!!(x), 0)
+#else
+#define LIKELY(x) (x)
+#define UNLIKELY(x) (x)
+#endif
+
+/*
+ * TRIGGER_ABORT --
+ *  Abort the program.
+ *
+ * When unit testing assertions we don't want to call __wt_abort, but we do want to track that we
+ * should have done so.
+ */
+#ifdef HAVE_UNITTEST_ASSERTS
+#define TRIGGER_ABORT(session, exp, ...)                                                       \
+    do {                                                                                       \
+        size_t _offset;                                                                        \
+        if ((session) == NULL) {                                                               \
+            __wt_errx(                                                                         \
+              session, "A non-NULL session must be provided when unit testing assertions");    \
+            __wt_abort(session);                                                               \
+        }                                                                                      \
+                                                                                               \
+        /*                                                                                     \
+         * Normally these are two errx messages, but we need to stitch them both together into \
+         * unittest_assert_msg. To do so insert the second message immediately after the first \
```

Comment:
np: Add a fullstop at the end of this sentence.

Reply:
Fixed in https://github.com/wiredtiger/wiredtiger/pull/8667/commits/5e90478e84e47cf776d3e84a6fce372b2f17396f 👍